src/systemd-status-mail

```diff
@@ -50,7 +52,16 @@ EOF
 	if [ -n "${RELAYHOST}" ]; then
 	    RELAY="-Ssmtp=${RELAYHOST}"
 	fi
-	mailx $MAILX_OPTIONS -Ssendwait -s "$SERVICE ($HOSTNAME)" -r "$FROM" "$RELAY" "$TO" <<EOF
+	if [ -n "${MAILX_AUTH_USER}" ]; then
+	    ENV="smtp-auth-user=${MAILX_AUTH_USER}"
+	fi
+	if [ -n "${MAILX_AUTH_PASSWORD}" ]; then
+	    ENV="$ENV smtp-auth-password=${MAILX_AUTH_PASSWORD}"
+	fi
+	if [ -n "$ENV" ]; then
+	    ENV="env $ENV"
```

Comment:
You must not use env here or the password is publically visible in `ps`

Reply:
The problem is that `export` can't be used here because `smtp-auth-password` and `smtp-auth-user` are not valid Shell variables due to the `-` in the name.

The correct solution (but that didn't gather a lot of love when we discussed this last time) would be to store the values in a mailx configuration file and use that one. That file could be temporary (just used by this script) or a permanent one - then the user could also customize it further.

Reply:
The generic and sensitive data is stored into and read from a temporary mailrc file now.